### DISCUSSION TITLE
tweak to the filter for image upload

### DISCUSSION THREAD

**Discussion by gavinsboyle** [sly9/VoyagerTelegramBot#33] [Ideas]
hi guys 
currently we have an exposure time filter for images - presumably to prevent pointing or flat frames being uploaded.
an issue i have is that with my photometry work sometimes we do have shorter exposures - i might have flat frames that get up to 10 seconds long and have science exposures at 9 seconds.

might it be possible to have an additional filter (which if left blank does nothing) which would be "image title must contain <>" - eg my science frames have LIGHT in their title and the flats have FLAT...(pointing exposures have neither...)

best
gavin

**Comment by sly9** [comment]
This worths some investigation — by default the file name does contain ‘LIGHT’, ‘BIAS’, ‘FLAT’, however user can customize the file naming pattern and might skip the ‘LIGHT’ keyword.

Another direction we can try is — we are getting the jpg image from ‘NewJPGReady’ event, and there’s another event called ‘NewFITReady’ which contains the image type(light/bias/dark/flat). If we can reliably link both events, this is doable

**Reply by bigpizza** [reply]
Hi Gavin,

I changed the criteria of sending image message. Now, if a JPG image (preview generated by Voyager) has it's corespoonding FIT image categorized as LIGHT SHOT, the preview will be sent to client regardless its exposure time. You could pull the latest code or wait for our next binary version.

Thanks for the suggestion and clear skies!

Kun
